platform/darwin/src/MGLGeoJSONSource.mm

```diff
@@ -28,6 +30,7 @@ - (instancetype)initWithIdentifier:(NSString *)identifier geoJSONData:(NSData *)
     {
         _geoJSONData = data;
         _options = options;
+        [self commonInit];
```

Comment:
That would be a nice refactor. However, `commonInit` relies on having properties like `URL` and `geoJSONData` and `features` set before it runs. If `MGLSource`'s `initWithIdentifier:` calls `commonInit` before the subclass init can set the requisite properties, then unexpected results will occur. I can imagine some ways to work around this but I think the disparate calls may actually be the best way to go, for now.